### ISSUE TITLE
Nested group tab completetion incorrect

### ISSUE THREAD

**Issue by nicholascapo** [pallets/click#520]
This problem appeared in version v6.0, it does not exist in v5.1.

The below example program uses nested groups:
```
cli -> a -> b
cli -> c -> d
```

Under click v5.1:
```bash
$ cli a <tab><tab> b
```

Under click v6.x:
```bash
$ eval "$(_CLI_COMPLETE=source /home/nicholas/cli)"

$ cli a <tab><tab>
a  c  <---- should be 'b'
```

If I then type `a`:
```
$ cli a a <tab><tab> b <---- correct except for the duplicated 'a'
```

Example program:
```python
#!/usr/bin/env python

import click


@click.group()
def cli():
    pass


@cli.group()
def a():
    pass


@a.command()
def b():
    pass


@cli.group()
def b():
    pass


@b.command()
def c():
    pass


if __name__ == '__main__':
    cli()



```

**Comment by laurrentt** [comment]
I've noticed the same thing. Rollbacking to 5.1 in the meantime.

**Comment by untitaker** [comment]
Very likely fixed by https://github.com/pallets/click/pull/545

**Comment by nicholascapo** [comment]
Confirmed fixed in version `6.6` :+1:
